Source/AwsCommonRuntimeKit/crt/AwsError.swift

```diff
@@ -2,8 +2,19 @@
 //  SPDX-License-Identifier: Apache-2.0.
 import AwsCCommon
 
-public enum AwsError: Error {
-  case fileNotFound(String)
-  case memoryAllocationFailure
-  case stringConversionError(UnsafePointer<aws_string>?)
+public struct AWSError {
+
+    public let errorCode: Int32
+
+    public let errorMessage: String?
+
+    public init(errorCode: Int32) {
+        self.errorCode = errorCode
+        let stringPtr = aws_error_str(errorCode)
+        if let stringPtr = stringPtr {
```

Comment:
What foul magic is this? Wouldn't this just be `if stringPtr`?

Reply:
Or even 
```swift
self.errorMessage = stringPtr ? String(cString: stringPtr) : nil
```

Reply:
this is just poor form, I will fix

Reply:
fixed, just bad swift on my part